receiver/redisreceiver/info.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redisreceiver
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+)
+
+// A map of the INFO data returned from Redis.
+type info map[string]string
+
+// Builds protobuf metrics from the combination of a metrics map
+// (INFO from Redis) and redisMetrics (built at startup). These are the fixed,
+// non keyspace metrics. Returns a list of parsing errors, which can be treated
+// like warnings.
+func (i info) buildFixedProtoMetrics(
+	metrics []*redisMetric,
+	t time.Time,
+) ([]*metricspb.Metric, []error) {
+	var warnings []error
+	var protoMetrics []*metricspb.Metric
+	for _, redisMetric := range metrics {
+		strVal, ok := i[redisMetric.key]
+		if !ok {
+			warnings = append(warnings, fmt.Errorf("info key not found: %v", redisMetric.key))
+			continue
+		}
+		if len(strVal) == 0 {
```

Comment:
nit: this repo in general compares against ""

Reply:
Noted, fixed.